libclc/CMakeLists.txt

```diff
@@ -449,6 +451,10 @@ foreach( t ${LIBCLC_TARGETS_TO_BUILD} )
       "+cl_khr_3d_image_writes")
     if(supports_generic_addrspace)
       string( APPEND CL_3_0_EXTENSIONS ",+__opencl_c_generic_address_space" )
+    elseif( ${ARCH} STREQUAL "x86_64")
+      string( APPEND CL_3_0_EXTENSIONS ",-__opencl_c_generic_address_space" )
```

Comment:
Do we really need to `-` the `__opencl_c_generic_address_space` if we're not explicitly adding it, above?

It seems a bit convoluted to test `x86_64` above, set `supports_generic_addrspace` = `FALSE`, then have to test `x86_64` again and do what `!supports_generic_addrspace` is implicitly trying to do.

Reply:
The `__opencl_c_generic_address_space` seems to be enabled by default for x86, so we need to manually disable it (and the extensions that require it). I've added the second check for `x86_64` in order to avoid affecting other targets, even though those extensions should already be disabled if `__opencl_c_generic_address_space` is disabled.

Reply:
Ah okay fair enough. I think it's worth pursuing this option for all targets. It definitely can't hurt to explicitly disable `__opencl_c_generic_address_space` for all architectures for which we set `supports_generic_addrspace` to `FALSE`. As for the others...

I see that `__opencl_c_pipes` is specified as requiring `__openclc_generic_address_space` :

```
OpenCL C compilers that define the feature macro __opencl_c_pipes must also define the feature macro __opencl_c_generic_address_space.
```

Same for `__opencl_c_device_enqueue`:

```
OpenCL C compilers that define the feature macro __opencl_c_device_enqueue must also define __opencl_c_generic_address_space and __opencl_c_program_scope_global_variables feature macros.
```

So, something like:

``` cmake
if(supports_generic_addrspace)
else()
  `-__openclc_generic_address_space`
  `-__opencl_c_pipes`
  `-__opencl_c_device_enqueue`
endif()
```

Looks correct for all targets (well, for us, just nvidia and native cpu at this stage)

Reply:
Right, I've removed the second check for `x86_64` and added a comment to explain why we disable the extensions